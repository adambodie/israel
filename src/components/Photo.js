import React from 'react';

const Photo = ({farm, server, id, secret, title, index}) => (
  <div>
        <img src={`https://farm${farm}.staticflickr.com/${server}/${id}_${secret}_z.jpg`} alt={title}/>
		<p>{title}</p>
  </div>
);

export default Photo;
