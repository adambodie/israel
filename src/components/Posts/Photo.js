import React from 'react';
import { imageLink } from '../../data/articles';

const Photo = ({id, title, index, length}) => (
	<div>
		<img src={`${imageLink}/IMG_${id}.JPG`} alt={title}/>
		<h3>{title}</h3>
		<p>{index} of {length}</p>
  </div>
);

export default Photo;
