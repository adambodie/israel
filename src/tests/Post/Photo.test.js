import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import Photo from '../../components/Posts/Photo';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Photo />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const tree = renderer.create(<Photo id="1" title="title" index="2" length="10"/>).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders a `photo`', () => {
	const wrapper = shallow(<Photo />);
    expect(wrapper.find('img').length).toEqual(1);
 });