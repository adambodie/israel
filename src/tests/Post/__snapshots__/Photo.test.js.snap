// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <img
    alt="title"
    src="https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/IMG_1.JPG"
  />
  <h3>
    title
  </h3>
  <p>
    2
     of 
    10
  </p>
</div>
`;
